File: sdks/ios/Sources/XMTPiOS/Libxmtp/GroupMembershipResult.swift
//
//  GroupMembershipResult.swift
//
//
//  Created by Naomi Plasterer on 3/10/25.
//

import Foundation

public struct GroupMembershipResult {
	var ffiGroupMembershipResult: FfiUpdateGroupMembershipResult

	public var addedMembers: [InboxId] {
		ffiGroupMembershipResult.addedMembers.map(\.key)
	}

	public var removedMembers: [InboxId] {
		ffiGroupMembershipResult.removedMembers
	}

	public var failedInstallationIds: [String] {
		ffiGroupMembershipResult.failedInstallations.map(\.toHex)
	}
}

File: sdks/android/jvm/src/main/java/org/xmtp/kotlin/libxmtp/GroupMembershipResult.kt
package org.xmtp.kotlin.libxmtp

import org.xmtp.kotlin.InboxId
import org.xmtp.kotlin.toHex
import uniffi.xmtpv3.FfiUpdateGroupMembershipResult

class GroupMembershipResult(
    private val ffiUpdateGroupMembershipResult: FfiUpdateGroupMembershipResult,
) {
    val addedMembers: List<InboxId>
        get() = ffiUpdateGroupMembershipResult.addedMembers.map { it.key }
    val removedMembers: List<InboxId>
        get() = ffiUpdateGroupMembershipResult.removedMembers
    val failedInstallationIds: List<String>
        get() = ffiUpdateGroupMembershipResult.failedInstallations.map { it.toHex() }
}
